base/common/python/pki/nssdb.py

```diff
@@ -1751,6 +1751,37 @@ def export_pkcs12(self, pkcs12_file,
         finally:
             shutil.rmtree(tmpdir)
 
+    def extract_ca_cert(self, ca_path, nickname):
+        tmpdir = tempfile.mkdtemp()
+
+        try:
+            p12_file = os.path.join(tmpdir, "sslserver.p12")
+            password = pki.generate_password()
+
+            # Build a chain containing the certificate we're trying to
+            # export. OpenSSL gets confused if we don't have a key for
+            # the end certificate: rh-bz#1246371
```

Comment:
I think we can actually prevent exporting keys. We're actually exporting _all_ certs here (not just the sslserver cert) because the initial cert this code is called from is a self-signed cert which lacks a signing CA. We can use cafiles to specify the server cert so it doesn't really matter.

Reply:
I removed this code that exported the CA certificate. I don't remove the leaf cert because the first step of pkispawn includes a self-signed certificate, which thus becomes a "CA" and a "leaf" certificate at the same time. We need it for trust to work, so I end up keeping everything exported. This is a excessive, but works given our blunt tools.